qualang_tools/video_mode/videomode.py

```diff
@@ -0,0 +1,452 @@
+                else:
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(fixed, value={parameter_value})"
+                    self.table[parameter_name]["type"] = float
+
+                self.table[parameter_name]["length"] = 0
+
+            elif isinstance(parameter_value, bool):
+                self.table[parameter_name][
+                    "declare_expression"
+                ] = f"declare(bool, value={parameter_value})"
+                self.table[parameter_name]["type"] = bool
+                self.table[parameter_name]["length"] = 0
+
+            elif isinstance(parameter_value, int):
+                self.table[parameter_name][
+                    "declare_expression"
+                ] = f"declare(int, value={parameter_value})"
+                self.table[parameter_name]["type"] = int
+                self.table[parameter_name]["length"] = 0
+
+            elif isinstance(parameter_value, (List, np.ndarray)):
+                if isinstance(parameter_value, np.ndarray):
+                    assert (
+                        parameter_value.ndim == 1
+                    ), "Invalid parameter type, array must be 1D."
+                    parameter_value = parameter_value.tolist()
+                assert all(
+                    isinstance(x, type(parameter_value[0])) for x in parameter_value
+                ), "Invalid parameter type, all elements must be of same type."
+                if isinstance(parameter_value[0], bool):
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(bool, value={parameter_value})"
+                elif isinstance(parameter_value[0], int):
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(int, value={parameter_value})"
+                elif isinstance(parameter_value[0], float):
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(fixed, value={parameter_value})"
+                self.table[parameter_name]["type"] = type(parameter_value)
+                self.table[parameter_name]["length"] = len(parameter_value)
+
+            else:
+                raise ValueError(
+                    "Invalid parameter type. Please use float, int or bool or list."
+                )
+
+    def declare_variables(self):
+        """
+        QUA Macro to create the QUA variables associated with the parameter table.
+        """
+        for parameter_name, parameter in self.table.items():
+            if parameter["declare_expression"] is not None:
+                exec(f"{parameter_name} = {parameter['declare_expression']}")
+                self.table[parameter_name]["var"] = eval(parameter_name)
+        pause()
+        if len(self.variables) == 1:
```

Comment:
This would be annoying from the user standpoint because he is getting the variables as `var1, var2, var3 = declare_variables()`. If the function returns a list of one element, then the user will need to write `var1 = declare_variables()[0]` no?

Reply:
The whole point of the class is to conceal the complexity of usage with natural calls. Although I would recommend the user to retrieve the QUA variables in the program through `video_mode["param_name"]` and not by declaring them as shown in Théo's version, but I figure it is a matter of taste. I would leave it as it is to enable both versions.